src/custom/state/claim/updater.tsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react'
+import { useBlockNumber } from '../application/hooks'
+import { SupportedChainId } from 'constants/chains'
+import { supportedChainId } from 'utils/supportedChainId'
+import { useActiveWeb3React } from 'hooks/web3'
+import { fetchClaims, useClaimDispatchers } from './hooks'
+
+const SUPPORTED_CLAIM_CHAINS = [SupportedChainId.MAINNET, SupportedChainId.XDAI]
+
+export default function Updater() {
+  const { account, chainId } = useActiveWeb3React()
+  const blockNumber = useBlockNumber()
+  const { setHasClaimsOnOtherChains } = useClaimDispatchers()
+
+  useEffect(() => {
+    if (!account) return
+
+    const supportedChain = supportedChainId(chainId)
+    SUPPORTED_CLAIM_CHAINS.forEach((chain) => {
+      if (supportedChain === chain) return
+      // Chain at this point is a supported chain
+      fetchClaims(account, chain)
+        .then((userClaims) => {
+          console.debug('[AvailbleClaimsUpdater]::Claim on', chain, userClaims)
+          setHasClaimsOnOtherChains({ chain, hasClaims: true })
+        })
+        .catch((error) => {
+          console.error('[AvailbleClaimsUpdater]::Error for chain', chain, error)
+          setHasClaimsOnOtherChains({ chain, hasClaims: false })
+        })
+    })
+  }, [account, blockNumber, chainId, setHasClaimsOnOtherChains])
```

Comment:
isn't this a performance issue if you add this check on every block? 

This is likely something we would want to check once, right? we then just make when the user claims, we take it into account

Reply:
yeah blockNumber here was for testing, will remove

Reply:
Okay, lets make sure we don't add more performance issues 👌